### ISSUE TITLE
bug: cannot disable messages

### ISSUE THREAD

**Issue by dudicoco** [folke/noice.nvim#941]
### Did you check docs and existing issues?

- [X] I have read all the noice.nvim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of noice.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

v0.9.5

### Operating system/version

MacOs Sonoma 14.4.1

### Describe the bug

Setting messages to `enabled = false` does nothing, I still get message notifications.

### Steps To Reproduce

1. Disable messages in config

### Expected Behavior

Message notifications should not show.

### Repro

```lua
return {
	{
		"folke/noice.nvim",
		event = "VeryLazy",
		dependencies = {
			"MunifTanjim/nui.nvim",
			"rcarriga/nvim-notify",
		},
		config = function()
			require("noice").setup({
				messages = {
					enabled = false,
				},
			})
		end,
	},
}
```

**Comment by eliasnorrby** [comment]
TL;DR: try:

```lua
messages = {
  enabled = false,
},
notify = {
  enabled = false,
}
```

Your reproduction lacks an example of a message you expect to not show. 

For me, disabling `messages` makes it so that e.g. running:

```vim
:lua print('hello')
```

produces a message shown in the `cmdline`, i.e. not handled by `noice`.

My impression is that [things like the LSP will call `vim.notify` directly](https://github.com/folke/noice.nvim/wiki/A-Guide-to-Messages#vimnotify). If you want to get rid of those, you need to disable `notify` too.

**Comment by folke** [comment]
that just disabled the messages integrations.
Notifications are a NEovim feature, which would of cours still go through. That's not noice

**Comment by dudicoco** [comment]
disabling both messages and notify fixed the problem, thanks!
